scripts/install_dependencies/install_openvino_dependencies.sh

```diff
@@ -16,260 +16,344 @@
+                -e DEBIAN_FRONTEND=noninteractive \
+                $image \
+                bash /scripts/${0##*/} $opt --keepcache
+            echo "||"
+            echo "|| Completed: $image / '$opt'"
+            echo "||"
+        done
     done
-}
+    echo "Self test finished, to remove temporary docker volumes run:
+        'docker volume rm yum-cache apt-cache'"
+    exit 0
+fi
 
-yes_or_no_gst_bad_ugly() {
-    if [ "$params" == "-y" ]; then
-        return 0
-    fi
+#===================================================================================================
+# OS detection
 
-    while true; do
-        read -p "Add third-party RPM Epel, Nux, Fusion, Forensics repositories and install dependencies for GStreamer Bad & Ugly Plugins (y/n): " yn
-        case $yn in
-            [Yy]*) return 0  ;;
-            [Nn]*) return  1 ;;
-        esac
-    done
-}
-
-if [ -f /etc/lsb-release ]; then
-    # Ubuntu
-    echo
-    echo "This script installs the following OpenVINO 3rd-party dependencies:"
-    echo "  1. GTK+, FFmpeg and GStreamer libraries used by OpenCV"
-    echo "  2. libusb library required for Myriad plugin for Inference Engine"
-    echo "  3. build dependencies for OpenVINO samples"
-    echo "  4. build dependencies for GStreamer Plugins"
-    echo
-    PKGS=(
-        cpio
-        build-essential
-        cmake
-        libusb-1.0-0-dev
-        libdrm-dev
-        libgstreamer1.0-0
+if [ "$os" == "auto" ] ; then
+    os=$( . /etc/os-release ; echo "${ID}${VERSION_ID}" )
+    case $os in
+        centos7|ubuntu18.04|ubuntu20.04) echo "Detected OS: ${os}" ;;
+        *) echo "Unsupported OS: ${os:-detection failed}" >&2 ; exit 1 ;;
+    esac
+fi
+
+#===================================================================================================
+# Collect packages
+
+extra_repos=()
+
+if [ "$os" == "ubuntu18.04" ] ; then
+
+    pkgs_opencv_req=(libgtk-3-0)
+    pkgs_python=(python3)
+    pkgs_dev=(cmake g++ gcc libc6-dev make)
+    pkgs_myriad=(libusb-1.0-0)
+    pkgs_installer=(cpio)
+    pkgs_opencv_opt=(
+        gstreamer1.0-plugins-bad
         gstreamer1.0-plugins-base
         gstreamer1.0-plugins-good
+        gstreamer1.0-plugins-ugly
+        libavcodec57
+        libavformat57
+        libavresample3
+        libavutil55
+        libgstreamer1.0-0
+        libswscale4
+    )
+    pkgs_dlstreamer=(
+        ffmpeg
+        flex
+        gstreamer1.0-alsa
         gstreamer1.0-plugins-bad
+        gstreamer1.0-plugins-base
+        gstreamer1.0-plugins-good
+        gstreamer1.0-plugins-ugly
         gstreamer1.0-vaapi
+        libfaac0
+        libfluidsynth1
+        libgl-dev
+        libglib2.0
+        libgstreamer1.0-0
+        libnettle6
+        libtag-extras1
+        python3-gi
+    )
+
+elif [ "$os" == "ubuntu20.04" ] ; then
+
+    pkgs_opencv_req=(libgtk-3-0)
+    pkgs_python=(python3)
+    pkgs_dev=(cmake g++ gcc libc6-dev make)
```

Comment:
Add libblas-dev liblapack-dev gfortran to dev component

Reply:
Which component requires these packages? I can not find them in current version of the script.

Reply:
it needs for building pot pypi dependencies on ubuntu20

Reply:
Then maybe it would be better to add separate POT component?

Reply:
agree